verifier/src/lib.rs

```diff
@@ -3,10 +3,222 @@
 #![doc = include_str!("../README.md")]
 #![deny(missing_docs, missing_debug_implementations, unsafe_code)]
 
+use std::fmt::{Debug, Formatter};
+
+type Result<T> = std::result::Result<T, VerificationError>;
+
+#[derive(Debug, Eq, PartialEq)]
+/// Failed to verify: {0}.
+pub struct VerificationError(String);
```

Comment:
In general I dislike the `Error(String)` pattern, since I can't do anything but print it out (e.g. what if the `TcbInfo` I've been given is simply out-of-date and I need to fetch a new one). This also means we can never communicate securely between a hardware element that doesn't support alloc and an enclave.

My preference is to have this be an associated type on `Verifier` (which is what I did in `mc-attest-verifier`)

Reply:
> My preference is to have this be an associated type on Verifier (which is what I did in mc-attest-verifier)

Where is that at, I did a quick scrape and only saw it on some hex encode/decode stuff

Reply:
Sorry, this was in my initial stab at improving error logging: https://github.com/jcape/mobilecoin/blob/92d200058ee597fbd94dac2d002e2439a95b2636/attest/verifier/src/report_body.rs#L102

Reply:
using associated error type now.
Currently there is a generic VerificationError. My thought is as we pull pieces over from the Epid logic we can expand the enum that it will become.